# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build freetype-build sdl-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --print-requires sdl-build | grep -q egl-opengl-stubs-build && GL=1

# configure (ac-2.63; am-1.11; lt-2.2.6)
libtoolize -c -f; aclocal -I acinclude; autoconf; automake -a -c
PKG_CONFIG_PATH=/dfb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/dfb/lib ./configure --disable-static --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
test $GL && install .libs/glfont $DESTDIR/dfb/bin
install .libs/showfont $DESTDIR/dfb/bin
install -d $DESTDIR/dfb/include/SDL
install -m 644 SDL_ttf.h $DESTDIR/dfb/include/SDL
install -d $DESTDIR/dfb/lib
install .libs/libSDL_ttf-1.2.so.0 $DESTDIR/dfb/lib
ln -sf libSDL_ttf-1.2.so.0 $DESTDIR/dfb/lib/libSDL_ttf.so
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 SDL_ttf.pc $DESTDIR/dfb/lib/pkgconfig

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/sdl_ttf-build.pc << EOF
Name: SDL_ttf
Version: 2.0.10
Description: TrueType font library for SDL
Requires: $REQUIRES

devel=\\
/dfb/include/SDL/SDL_ttf.h \\
/dfb/lib/libSDL_ttf.so \\
/dfb/lib/pkgconfig/SDL_ttf.pc

exec=\\
EOF
test $GL && cat >> $DESTDIR/dfb/share/pkgconfig/sdl_ttf-build.pc << EOF
/dfb/bin/glfont \\
EOF
cat >> $DESTDIR/dfb/share/pkgconfig/sdl_ttf-build.pc << EOF
/dfb/bin/showfont \\
/dfb/lib/libSDL_ttf-1.2.so.0
EOF

exit

# patch
--- SDL_ttf-2.0.10.orig/Makefile.am
+++ SDL_ttf-2.0.10/Makefile.am
@@ -1,5 +1,7 @@
 # Makefile.am for the SDL truetype font loading library and viewer
 
+AUTOMAKE_OPTIONS = foreign
+
 lib_LTLIBRARIES = libSDL_ttf.la
 
 libSDL_ttfincludedir = $(includedir)/SDL
--- SDL_ttf-2.0.10.orig/configure.in
+++ SDL_ttf-2.0.10/configure.in
@@ -1,5 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README)
+AC_CONFIG_MACRO_DIR(acinclude)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -11,8 +12,8 @@
 # if backwards compatibility has been broken,
 # set BINARY_AGE and INTERFACE_AGE to 0.
 
-MAJOR_VERSION=2
-MINOR_VERSION=0
+MAJOR_VERSION=1
+MINOR_VERSION=2
 MICRO_VERSION=10
 INTERFACE_AGE=0
 BINARY_AGE=10
@@ -116,9 +117,12 @@
 AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
 no_freetype=""
 if test "$FREETYPE_CONFIG" = "no" ; then
+    PKG_CHECK_MODULES([FT2], [freetype2], [
+    CFLAGS="$CFLAGS $FT2_CFLAGS"
+    LIBS="$LIBS $FT2_LIBS"],
     AC_MSG_ERROR([
 *** Unable to find FreeType2 library (http://www.freetype.org/)
-])
+]))
 else
     CFLAGS="$CFLAGS `$FREETYPE_CONFIG $freetypeconf_args --cflags`"
     LIBS="$LIBS `$FREETYPE_CONFIG $freetypeconf_args --libs`"
@@ -156,14 +160,11 @@
         ;;
     *)
         MATHLIB="-lm"
-        AC_PATH_X
-        AC_PATH_XTRA
-        if test x$have_x = xyes; then
-            CFLAGS="$CFLAGS $X_CFLAGS"
-            SYS_GL_LIBS="$X_LIBS -lGL"
-        else
+        PKG_CHECK_MODULES([SYS_GL], [gl], [
+            CFLAGS="$CFLAGS $SYS_GL_CFLAGS"
+        ], [
             SYS_GL_LIBS="-lGL"
-        fi
+        ])
         ;;
 esac
 AC_MSG_CHECKING(for OpenGL support)

# source
https://www.libsdl.org/projects/SDL_ttf/release/SDL_ttf-2.0.10.tar.gz
